### DISCUSSION TITLE
Cannot load AnimeIVTC

### DISCUSSION THREAD

**Discussion by Daud01** [staxrip/staxrip#1255] [Q&A] [answered]
Hi! I've started using staxrip 2 days ago so go easy on me. 
I'm trying to inverse telecine a 3:2 ntsc stream and every time I try to load up AnimeIVTC with any of the selected modes, it throws this error:

Script Error

System exception - Illegal Instruction
(C:\Users\lenovo\Desktop\StaxRip-v2.28.0-x64\Apps\Plugins\AVS\AnimeIVTC\AnimeIVTC.avsi, line 196)

The "line 196" in question:
`edeintn= ((usecbob && cbob < 6) || (usebbob && bbob < 6) || ((usefm || usehfm) && precision < 6)) || autoAssuf ? defined(ediext) ? ediext : ediandnn != 2 || ediandnn != -2 ? ediandnn < 0 ? iforbob.sh_Padding(2,4,2,4,threads=threads).nnedi3(-2,threads=threads).crop(2,4,-2,-4,true) : iforbob.nnedi3(-2,threads=threads) : nop() : nop()`

Attached sample file: https://mega.nz/file/JlFlGLxD#0QhTB8S2fD--wbfzWPUYkwoULvyNWv0SMAoVxUnEMrE

Can someone point out what's going wrong(it's probably my fault)? Any kind of help would be appreciated.

**Comment by Dendraspis** [comment]
Can you also post the filter call from your script?

**Reply by Daud01** [reply]
Is this it ?
```
AddAutoloadDir("C:\Users\lenovo\Desktop\StaxRip-v2.36.0-x64\Apps\FrameServer\AviSynth\plugins")
LoadPlugin("C:\Users\lenovo\Desktop\StaxRip-v2.36.0-x64\Apps\Plugins\AVS\Bwdif\BWDIF.dll")
Import("C:\Users\lenovo\Desktop\StaxRip-v2.36.0-x64\Apps\Plugins\AVS\DAA3Mod\daa3mod.avsi")
LoadPlugin("C:\Users\lenovo\Desktop\StaxRip-v2.36.0-x64\Apps\Plugins\AVS\Decomb\Decomb.dll")
LoadPlugin("C:\Users\lenovo\Desktop\StaxRip-v2.36.0-x64\Apps\Plugins\AVS\EEDI3\eedi3.dll")
LoadPlugin("C:\Users\lenovo\Desktop\StaxRip-v2.36.0-x64\Apps\Plugins\AVS\GRunT\grunt.dll")
LoadPlugin("C:\Users\lenovo\Desktop\StaxRip-v2.36.0-x64\Apps\Plugins\AVS\JPSDR\Plugins_JPSDR.dll")
LoadPlugin("C:\Users\lenovo\Desktop\StaxRip-v2.36.0-x64\Apps\Plugins\AVS\masktools2\masktools2.dll")
LoadPlugin("C:\Users\lenovo\Desktop\StaxRip-v2.36.0-x64\Apps\Plugins\AVS\mvtools2\mvtools2.dll")
Import("C:\Users\lenovo\Desktop\StaxRip-v2.36.0-x64\Apps\Plugins\AVS\QTGMC\QTGMC.avsi")
LoadPlugin("C:\Users\lenovo\Desktop\StaxRip-v2.36.0-x64\Apps\Plugins\AVS\ReduceFlicker\ReduceFlicker.dll")
LoadPlugin("C:\Users\lenovo\Desktop\StaxRip-v2.36.0-x64\Apps\Plugins\AVS\RgTools\RgTools.dll")
LoadPlugin("C:\Users\lenovo\Desktop\StaxRip-v2.36.0-x64\Apps\Plugins\AVS\RT_Stats\RT_Stats_x64.dll")
Import("C:\Users\lenovo\Desktop\StaxRip-v2.36.0-x64\Apps\Plugins\AVS\Scripts\SharpenersPack.avsi")
Import("C:\Users\lenovo\Desktop\StaxRip-v2.36.0-x64\Apps\Plugins\AVS\Scripts\ExTools.avsi")
Import("C:\Users\lenovo\Desktop\StaxRip-v2.36.0-x64\Apps\Plugins\AVS\Scripts\ResizersPack.avsi")
Import("C:\Users\lenovo\Desktop\StaxRip-v2.36.0-x64\Apps\Plugins\AVS\Srestore\Srestore.avsi")
LoadPlugin("C:\Users\lenovo\Desktop\StaxRip-v2.36.0-x64\Apps\Plugins\AVS\TDeint\TDeint.dll")
LoadPlugin("C:\Users\lenovo\Desktop\StaxRip-v2.36.0-x64\Apps\Plugins\AVS\TIVTC\TIVTC.dll")
LoadPlugin("C:\Users\lenovo\Desktop\StaxRip-v2.36.0-x64\Apps\Plugins\AVS\TMM2\TMM2.dll")
LoadPlugin("C:\Users\lenovo\Desktop\StaxRip-v2.36.0-x64\Apps\Plugins\AVS\vinverse\vinverse.dll")
LoadPlugin("C:\Users\lenovo\Desktop\StaxRip-v2.36.0-x64\Apps\Plugins\AVS\yadifmod2\yadifmod2.dll")
Import("C:\Users\lenovo\Desktop\StaxRip-v2.36.0-x64\Apps\Plugins\AVS\Zs_RF_Shared\Zs_RF_Shared.avsi")
Import("C:\Users\lenovo\Desktop\StaxRip-v2.36.0-x64\Apps\Plugins\AVS\AnimeIVTC\AnimeIVTC.avsi")
LoadPlugin("C:\Users\lenovo\Desktop\StaxRip-v2.36.0-x64\Apps\Plugins\Dual\ffms2\ffms2.dll")
tcFile = "C:\Users\lenovo\Desktop\Sample_temp\Sample_timestamps.txt" # timestamps file path
Exist(tcFile) ? FFVideoSource("C:\Users\lenovo\Desktop\Sample.mkv", cachefile="C:\Users\lenovo\Desktop\Sample_temp\temp.ffindex", timecodes=tcFile) : FFVideoSource("C:\Users\lenovo\Desktop\Sample.mkv", cachefile="C:\Users\lenovo\Desktop\Sample_temp\temp.ffindex")
Crop(12, 0, -4, -0)
AnimeIVTC(mode=1, omode=1)
```

**Reply by Dendraspis** [reply]
No exactly, but that's even better.  😁 

But using `AnimeIVTC(mode=1, omode=1)` works for me without issues:
![image](https://github.com/staxrip/staxrip/assets/50659978/11fa0020-02fa-4413-a024-08b79c59c012)

**Reply by Daud01** [reply]
This is what mine looks like 😓

![Capture](https://github.com/staxrip/staxrip/assets/95249793/3a99216f-6591-40cd-95bb-0c53b586b189)

**Reply by Dendraspis** [reply]
I have also found an `Interlaced TFF MPEG` file, works without issues, so I sadly can't help. It guess it is a hardware dependency, that you don't fulfill.  😕 

<img src="https://github.com/staxrip/staxrip/assets/50659978/e8f3d624-1bc6-4663-8fc7-cf60a01efc01" width=700>

**Reply by Daud01** [reply]
Can you try this sample?

https://mega.nz/file/JlFlGLxD#0QhTB8S2fD--wbfzWPUYkwoULvyNWv0SMAoVxUnEMrE

**Reply by Dendraspis** [reply]
Yes, and it works:

<img width=700 src="https://github.com/staxrip/staxrip/assets/50659978/ac917b09-0114-4f3e-a2f8-8b97afbb5c8a">

**Reply by Daud01** [reply]
Have you changed anything related to video "demux" or timestamp extraction in the options toolbar?

**Reply by Dendraspis** [reply, marked as answer]
Everything should be default, meaning `Video demuxing` turned off and `Extract timestamps` set to `VFR only`, but that shouldn't matter.

Because the calling of `nnedi3` or `nnedi3cl` I guess it's related to too old or wrong GPU drivers or not supported GPU.

**Reply by Daud01** [reply]
> Because the calling of nnedi3 or nnedi3cl I guess it's related to too old or wrong GPU drivers or not supported GPU.

Guess I'm stuck with `TIVTC` instead 😔. Thanks for the help anyways.
One last thing, having looked at the source, can you point towards any safe denoising/aa filters that I can use after TIVTC?. Can we specify them to filter after the inverse telecine process? I tried some but they always get applied before `TIVTC`.

**Reply by Dendraspis** [reply]
I don't understand. You define the order, either through the code itself or through the order of the filters (via drag & drop).

**Reply by Daud01** [reply]
Sorry upon rechecking I applied the wrong filter after `TIVTC` , which resulted in really bad stair stepping.
